### ISSUE TITLE
[Bug]: Invalid prop `renderIcon` warning in console for menuItemSelectable

### ISSUE THREAD

**Issue by OmranAbazid** [carbon-design-system/carbon#14521]
### Package

@carbon/react

### Browser

Chrome

### Package version

V1.36

### React version

v18.2

### Description

When trying to render `MenuItemSelectable` Component inside a `Menu` component, i get the following warning:

```
Warning: Failed prop type: Invalid prop `renderIcon` supplied to `ForwardRef(MenuItem)`, expected one of type [function, object].
    at MenuItem (https://6r3y3j.csb.app/node_modules/@carbon/react/es/components/Menu/MenuItem.js:42:23)
    at MenuItemSelectable (https://6r3y3j.csb.app/node_modules/@carbon/react/es/components/Menu/MenuItem.js:209:25)
    at ul
    at li
    at MenuItemGroup (https://6r3y3j.csb.app/node_modules/@carbon/react/es/components/Menu/MenuItem.js:273:24)
    at ul
    at Menu (https://6r3y3j.csb.app/node_modules/@carbon/react/es/components/Menu/Menu.js:28:18)
    at div
    at App
```

### Reproduction/example

https://codesandbox.io/s/romantic-ace-6r3y3j?file=/index.js:240-244

### Steps to reproduce

- Place `MenuItemSelectable`  inside a Menu component 
- Open the console to see the warning

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
Thanks for pointing this out, we'll take a look. I'm not sure what's going on as MenuItem accepts a `renderIcon`, and `MenuItemSelectable` appears to be passing the Checkmark icon properly.

https://github.com/carbon-design-system/carbon/blob/d88958772d7287cc99bdd592b85fdd9ea19c3af9/packages/react/src/components/Menu/MenuItem.js#L251

**Comment by preetibansalui** [comment]
I can't reproduce the issue and it may have been fixed by [#15177](https://github.com/carbon-design-system/carbon/issues/15177). If this continues to be an issue for you please open a new issue.
